crates/ruff_server/src/server/api/notifications/did_open_notebook.rs

```diff
@@ -0,0 +1,65 @@
+use crate::edit::NotebookDocument;
+use crate::server::api::diagnostics::publish_diagnostics_for_document;
+use crate::server::api::LSPResult;
+use crate::server::client::{Notifier, Requester};
+use crate::server::Result;
+use crate::session::Session;
+use anyhow::anyhow;
+use lsp_server::ErrorCode;
+use lsp_types as types;
+use lsp_types::notification as notif;
+
+pub(crate) struct DidOpenNotebook;
+
+impl super::NotificationHandler for DidOpenNotebook {
+    type NotificationType = notif::DidOpenNotebookDocument;
+}
+
+impl super::SyncNotificationHandler for DidOpenNotebook {
+    fn run(
+        session: &mut Session,
+        notifier: Notifier,
+        _requester: &mut Requester,
+        types::DidOpenNotebookDocumentParams {
+            notebook_document:
+                types::NotebookDocument {
+                    uri,
+                    version,
+                    cells,
+                    metadata,
+                    notebook_type,
+                },
+            cell_text_documents,
+        }: types::DidOpenNotebookDocumentParams,
+    ) -> Result<()> {
+        if notebook_type != "jupyter-notebook" {
+            return Err(anyhow!(
+                "Tried to open an unsupported notebook type: {notebook_type}"
+            ))
+            .with_failure_code(ErrorCode::InvalidParams);
+        }
```

Comment:
I'm not sure if we should use this guard. What's your motivation for this? I don't think we have anything like this in `ruff-lsp`. Did you try this with any other editor that supports Jupyter Notebook and can run ruff?

Reply:
> Did you try this with any other editor that supports Jupyter Notebook and can run ruff?

As far as I know the only other major editor that supports Jupyter Notebook is PyCharm, and we haven't got that working with `ruff server` yet.

This is meant to be a quick sanity check to ensure that we're only working with Python notebooks, but it's not super necessary.

Reply:
I'm fine leaving it for now but it seems like a duplicate given the notebook filter the server has already applied.

Reply:
I'm fine with removing it